src/components/Button.tsx

```diff
@@ -1,62 +1,92 @@
+  onClick?: ((e: PressEvent) => void) | string;
+  /** Text to be shown via a tooltip when the user hovers over the button */
+  tooltip?: ReactNode;
 }
 
 export function Button(props: ButtonProps) {
-  const { onClick: onPress, disabled, endAdornment, menuTriggerProps, ...otherProps } = props;
+  const { onClick: onPress, disabled, endAdornment, menuTriggerProps, tooltip, ...otherProps } = props;
   const isDisabled = !!disabled;
   const ariaProps = { onPress, isDisabled, ...otherProps, ...menuTriggerProps };
   const { label, icon, variant = "primary", size = "sm", buttonRef } = ariaProps;
   const ref = buttonRef || useRef(null);
-  const { buttonProps, isPressed } = useButton(ariaProps, ref as RefObject<HTMLButtonElement>);
+  const { buttonProps, isPressed } = useButton(
+    {
+      ...ariaProps,
+      onPress: typeof onPress === "string" ? noop : onPress,
+      elementType: typeof onPress === "string" ? "a" : "button",
+    },
+    ref as RefObject<HTMLElement>,
+  );
   const { isFocusVisible, focusProps } = useFocusRing(ariaProps);
   const { hoverProps, isHovered } = useHover(ariaProps);
-  const { baseStyles, hoverStyles, disabledStyles, pressedStyles } = useMemo(() => getButtonStyles(variant, size), [
-    variant,
-    size,
-  ]);
+  const { baseStyles, hoverStyles, disabledStyles, pressedStyles } = useMemo(
+    () => getButtonStyles(variant, size),
+    [variant, size],
+  );
   const focusRingStyles = useMemo(() => (variant === "danger" ? Css.bshDanger.$ : Css.bshFocus.$), [variant]);
 
-  const button = (
-    <button
-      ref={ref}
-      {...buttonProps}
-      {...focusProps}
-      {...hoverProps}
-      css={{
-        ...Css.buttonBase.$,
-        ...baseStyles,
-        ...(isHovered && !isPressed ? hoverStyles : {}),
-        ...(isPressed ? pressedStyles : {}),
-        ...(isDisabled ? { ...disabledStyles, ...Css.cursorNotAllowed.$ } : {}),
-        ...(isFocusVisible ? focusRingStyles : {}),
-      }}
-    >
+  const buttonContent = (
+    <>
       {icon && <Icon xss={iconStyles[size]} icon={icon} />}
       {label}
       {endAdornment && <span css={Css.ml1.$}>{endAdornment}</span>}
-    </button>
+    </>
   );
+  const buttonAttrs = {
+    ref: ref as any,
+    ...buttonProps,
+    ...focusProps,
+    ...hoverProps,
+    css: {
+      ...Css.buttonBase.$,
+      ...baseStyles,
+      ...(isHovered && !isPressed ? hoverStyles : {}),
+      ...(isPressed ? pressedStyles : {}),
+      ...(isDisabled ? { ...disabledStyles, ...Css.cursorNotAllowed.$ } : {}),
+      ...(isFocusVisible ? focusRingStyles : {}),
+    },
+  };
 
-  // If we're disabled b/c of a non-boolean ReactNode, show it in a tooltip
-  if (isDisabled && typeof disabled !== "boolean") {
+  const button =
+    typeof onPress === "string" ? (
+      isAbsoluteUrl(onPress) ? (
+        <a {...buttonAttrs} href={onPress} className={navLink} target="_blank" rel="noreferrer noopener">
```

Comment:
This is making an assumption that every absolute URL should open in a new tab. Because of that assumption, I think we'd also be safe to always show `<Icon icon="linkExternal" />` within the button. I've added this in on the Storybook examples via `endAdornment`, though probably just have it done by default.